src/qat/purr/compiler/instructions.py

```diff
@@ -741,3 +752,137 @@ def calculate_duration(instruction, return_samples: bool = True):
         calc_sample = block_number * block_time
 
     return calc_sample
+
+
+class InstructionBlock:
+    def _validate_targets(self, targets, valid_types):
+        if targets is None:
+            targets = []
+        elif not isinstance(targets, List):
+            targets = [targets]
+
+        invalid_targets = [val for val in targets if not isinstance(val, valid_types)]
+        if any(invalid_targets):
+            invalid_targets_str = ",".join([str(val) for val in invalid_targets])
+            raise ValueError(f"Invalid targets for {type(self)}: {invalid_targets_str}")
+
+        return targets
+
+
+class QuantumInstructionBlock(InstructionBlock):
+    quantum_targets: List[QuantumComponent]
+    instructions: List[Instruction]
```

Comment:
Would it make sense to have a `duration` here as well, which is the summed duration of all the indivual instructions within the block?

Reply:
Coming right up! :slightly_smiling_face: :+1: